crates/torii/src/graphql/entity.rs

```diff
@@ -20,152 +18,73 @@ pub struct Entity {
-            .push_bind(partition_id);
-
-        if let Some(keys) = keys {
-            let keys_str = format!("{}%", keys.join(","));
-            builder.push(" AND keys LIKE ")
-                .push_bind(keys_str);
-        }
-
-        if let Some(after) = after {
-            let entity = entity_by_id(conn, after.0.to_string()).await?;
-            let created_at = entity.created_at.to_rfc3339_opts(SecondsFormat::Secs, true);
-            builder.push(" AND created_at > ")
-                .push_bind(created_at);
-        }
-
-        if let Some(before) = before {
-            let entity = entity_by_id(conn, before.0.to_string()).await?;
-            let created_at = entity.created_at.to_rfc3339_opts(SecondsFormat::Secs, true);
-            builder.push(" AND created_at < ")
-                .push_bind(created_at);
+impl ObjectTraitStatic for EntityObject {
+    fn new() -> Self {
+        Self {
+            field_type_mapping: IndexMap::from([
+                (String::from("id"), String::from("ID")),
+                (String::from("name"), String::from("String")),
+                (String::from("partitionId"), String::from("FieldElement")),
+                (String::from("keys"), String::from("String")),
+                (String::from("transactionHash"), String::from("FieldElement")),
+                (String::from("createdAt"), String::from("DateTime")),
+            ]),
         }
+    }
+    fn from(field_type_mapping: TypeMapping) -> Self {
+        Self { field_type_mapping }
+    }
+}
 
-        let order = match last {
-            Some(_) => "ASC",
-            None => "DESC",
-        };
-        builder.push(" ORDER BY created_at ").push(order);
+impl ObjectTraitInstance for EntityObject {
+    fn name(&self) -> &str {
+        "entity"
+    }
 
-        let limit = match first.or(last) {
-            Some(limit) => limit,
-            None => DEFAULT_LIMIT,
-        };
-        builder.push(" LIMIT ").push(limit.to_string());
+    fn type_name(&self) -> &str {
+        "Entity"
+    }
 
-        let entities: Vec<Entity> = builder.build_query_as().fetch_all(conn).await?;
+    fn field_type_mapping(&self) -> &TypeMapping {
+        &self.field_type_mapping
+    }
 
-        // TODO: hasPreviousPage, hasNextPage
-        let mut connection = Connection::new(true, true);
-        for entity in entities {
-            connection.edges.push(
-                Edge::new(
-                    OpaqueCursor(ID(entity.id.clone())),
-                    entity
-                )
-            );
-        }
-        Ok::<_, Error>(connection)
-    }).await
-}
+    fn field_resolvers(&self) -> Vec<Field> {
+        vec![
+            Field::new(self.name(), TypeRef::named_nn(self.type_name()), |ctx| {
+                FieldFuture::new(async move {
+                    let mut conn = ctx.data::<Pool<Sqlite>>()?.acquire().await?;
+                    let id = ctx.args.get("id").expect("id not found");
```

Comment:
expect like this i think panics? if so we should handle / log instead (unless it's impossible to hit)

Reply:
Yeah it shouldn't hit here, but will make it handle this better